Plugins/Json/src/JsonSurfacesReader.cpp

```diff
@@ -73,4 +73,27 @@ std::vector<std::shared_ptr<Acts::Surface>> JsonSurfacesReader::readVector(
   return surfaces;
 }
 
-}  // namespace ActsExamples
+std::vector<std::shared_ptr<Acts::JsonDetectorElement>>
+JsonSurfacesReader::readDetectorElements(const Options& options,
+                                         double thickness = 0.0) {
+  nlohmann::json j;
+  {
+    std::ifstream in(options.inputFile);
+    in >> j;
+  }
+
+  // Walk down the path to the surface entries
+  nlohmann::json jSurfaces = j;
+  for (const auto& jep : options.jsonEntryPath) {
+    jSurfaces = jSurfaces[jep];
```

Comment:
I don't understand what this does, honestly.

Reply:
This is some kind of path into the json file, in case the surface list is nested within an arbitrary json file...

Reply:
Not sure that clears it up for me. Why does this object need to decide this? Can't this be decided by the caller depending on where it's reading the surfaces from?

Reply:
So I didn't invent this, it was there before. But imagine you habe a json file like 
```json
{
	"foo": {
		"bar": [
			surface1,
			surface2,
		]
	}    
}
```

Then you could pass `["foo", "bar"]` to the reader. Not sure why it was there, but I decided to keep this pattern.

Reply:
I see, this explains why it's re-assigning the element.